models/models.go

```diff
@@ -809,30 +809,38 @@ func (m *RouteManager) ensureChallenges(route *Route, client *acme.Client, updat
 		"instance-id": route.InstanceId,
 	})
 
-	if len(route.ChallengeJSON) == 0 {
-		challenges, errs := client.GetChallenges(route.GetDomains())
-		if len(errs) > 0 {
-			err := fmt.Errorf("Error(s) getting challenges: %v", errs)
-			lsession.Error("get-challenges", err)
-			return err
-		}
+	if len(route.ChallengeJSON) > 0 {
+		lsession.Info("challenge-json-was-already-present")
+		return nil
+	}
 
-		var err error
-		route.ChallengeJSON, err = json.Marshal(challenges)
-		if err != nil {
-			lsession.Error("json-marshal-challenge", err)
-			return err
-		}
+	lsession.Info("get-challenges")
+	challenges, errs := client.GetChallenges(route.GetDomains())
+	if len(errs) > 0 {
+		err := fmt.Errorf("Error(s) getting challenges: %v", errs)
+		lsession.Error("get-challenges-err", err)
+		return err
+	}
 
-		if update {
-			err := m.db.Save(route).Error
-			lsession.Error("db-save-route-challenge", err)
-			return err
-		}
+	lsession.Info("json-marshal-challenge")
+	var err error
+	route.ChallengeJSON, err = json.Marshal(challenges)
+	if err != nil {
+		lsession.Error("json-marshal-challenge-err", err)
+		return err
+	}
+
+	if !update {
+		lsession.Info("not-updating")
 		return nil
 	}
 
-	return nil
+	lsession.Info("db-save-route-challenge")
+	err = m.db.Save(route).Error
+	if err != nil {
+		lsession.Error("db-save-route-challenge-err", err)
+	}
+	return err
```

Comment:
Could we insert a "challenge-ensure-success" message, or something of that ilk. Essentially a message we can scan for to see if everything was successful here?

Reply:
I added an `updated` message, and tweaked the function's other successful exit point to match.